types/streamx/index.d.ts

```diff
@@ -0,0 +1,309 @@
+// Type definitions for streamx 2.9
+// Project: https://github.com/streamxorg/streamx
+// Definitions by: Martin Heidegger <https://github.com/martinheidegger>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { EventEmitter } from 'events';
+import { Writable as NodeJSWritable, Readable as NodeJSReadable, Stream as NodeJSStream } from 'stream';
+
+export interface AbortSignalLike {
+    addEventListener: (type: 'abort', listener: (this: AbortSignalLike, event: any) => any) => void;
+}
+
+export type AnyStream = NodeJSStream | Stream;
+export type AnyWritable<TType = any, TMap = TType, TByte = TMap> =
+    | NodeJSWritable
+    | Writable<TType, TMap, TByte>
+    | Duplex<TType, TByte, TMap, TByte>;
+export type AnyReadable<TType = any, TMap = TType, TByte = TMap> = NodeJSReadable | Readable<TType, TMap, TByte>;
+
+export interface Callback {
+    (err?: Error | null): void;
+}
+
+export interface ResultCallback<T> {
+    (err?: Error | null, result?: T): void;
+}
+
+export interface StreamOptions<TStream extends Stream<TByteType>, TByteType = any> {
+    highWaterMark?: number;
+    byteLength?: ByteLengthFunction<TStream, TByteType>;
+    open?: (this: TStream, cb: Callback) => void;
+    destroy?: (this: TStream, cb: Callback) => void;
+    predestroy?: (this: TStream, cb: Callback) => void;
+    signal?: AbortSignalLike;
+}
+
+/* tslint:disable-next-line interface-over-type-literal */
```

Comment:
Would it make sense to add a comment for why this needs to be a type literal instead of an interface?

Reply:
Added in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49810/commits/2681fb33d831ab7fcac2e03b07932c6e714c4e2e